### PR TITLE
fix: clean up broken links

### PR DESCRIPTION
**Pull request by f-hollow**
## Description

Running lychee link checker on main returns a number of invalid links that should be cleaned up. Hopefully, some links can be updated during the PR review. Others can be left in `/.lycheeignore` permanently.

There is also an attempt to add a link to product selector. It was formerly committed or rendered with unsafe characters unencoded, such as `{`, `[`, `:`. For details on unsafe characters, see [RFC 1738](https://www.rfc-editor.org/rfc/rfc1738#section-2.2) > *2.2. URL Character Encoding Issues*.

## Testing

---

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [x] All CI checks (GH Actions) pass.
- [ ] Documentation is updated as needed.
- [ ] Tests are updated or added as necessary.
- [ ] Code is well-commented, especially in complex areas.
- [x] Git history is clean — commits are squashed to the minimum necessary.

### CODE REVIEW

File: content/blog/unleashing-the-power-of-openai-and-esp-box-a-guide-to-fusing-chatgpt-with-espressif-socs/index.md [hunk 1]
```diff
@@ -493,3 +493,5 @@ The integration of OpenAI’s ChatGPT with Espressif’s ESP-BOX has opened up n
     >}}
 
 Don’t forget to check out [Espressif Systems](https://www.espressif.com/)’ [GitHub ](https://github.com/orgs/espressif/repositories)repository for more open-source demos on [ESP-IoT-Solution](https://github.com/espressif/esp-iot-solution), [ESP-SR](https://github.com/espressif/esp-sr), and [ESP-BOX](https://github.com/espressif/esp-box). The source code for this project will be found [__*here*__ ](https://github.com/espressif/esp-box/tree/master/examples)*.* As part of our future plans, we aim to introduce a component for the OpenAI API that will offer a user-friendly functions.
+
+See also the [Product Selector](https://products.espressif.com/#/product-selector?names=&filter=%7B%22Series%22%3A%5B%22ESP32-S3%22%5D%7D).
```
Comment (f-hollow) [inline content/blog/unleashing-the-power-of-openai-and-esp-box-a-guide-to-fusing-chatgpt-with-espressif-socs/index.md:497]: @pedrominatel I added the encoded Product Selector link for testing purposes. Looks like there are no issues now.

Please let me know how and where to add this link in the article.

File: .lycheeignore [hunk 2]
```diff
@@ -37,3 +37,77 @@ socket://localhost
 
 # Bug on Lychee (see later)
 https://products.espressif.com/
+
+
+# List of invalid links as of 18.10.2024
+
+# ./content/blog/developing-user-interfaces-for-custom-boards-with-esp-microcontrollers/index.md
+https://github.com/espressif/esp-bsp/blob/master/SquareLine/boards/custom_waveshare_7inch/components/ws_7inch/CMakeLists.txt
```
Comment (f-hollow) [inline .lycheeignore:45]: @pedrominatel @georgik 

Can you please suggest a good approach to update or fix the invalid links to numerous Espressif repos?

Should the content in the respective articles be reviewed as well to see if it is still relevant?

File: config/_default/languages.en.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 languageCode = "en"
 languageName = "English"
 weight = 1
-title = "Developer Portal"
+title = "Espressif Developer Portal"
```
Comment (f-hollow) [inline config/_default/languages.en.toml:4]: @pedrominatel Suggest adding company name to website title.

Comment (pedrominatel) [reply config/_default/languages.en.toml:4]: Agree.

File: .lycheeignore (hunk 2 above)
Comment (pedrominatel) [reply .lycheeignore:45]: For the LVGL `SquareLine/boards/custom_waveshare_7inch`, we can change it to `SquareLine/boards/v8/custom_waveshare_7inch`.

We can change our GitHub repo links to the project repo root to find them in the new location. If the project does not exist, point to the Espressif Org on GitHub.

https://sqfmi.com/ works fine.

For third-party ones, please keep it on the ignore list.

**Review by pedrominatel — APPROVED** [review]
LGTM

**Comment by f-hollow** [comment]
@alisitszn I know that you maintain ESP-Modbus repo. I have a [link](https://github.com/espressif/developer-portal/blob/001f54f680027c89f0e648bd85931e36b57a230e/.lycheeignore#L6) to it that returns 404.

Can you please propose what to replace this link with?

**Comment by alisitsyn** [comment]
@f-hollow,

The documentation for the esp-modbus project is the same for all targets. the component has the dependencies to standard components such as UART driver, LWIP, MDNS, esp-timer. The [correct link to this documentation for esp-modbus v1.0.x](https://docs.espressif.com/projects/esp-modbus/en/latest).

**Review by alisitsyn — APPROVED** [review]
ESP-Modbus changes LGTM!

**Pull request merged**